packages/cli/src/collect/collect.js

```diff
@@ -174,7 +178,13 @@ async function startServerAndDetermineUrls(options) {
   const urls = urlsAsArray;
   if (!urls.length) {
     const maxNumberOfUrls = options.maxAutodiscoverUrls || Infinity;
-    urls.push(...server.getAvailableUrls().slice(0, maxNumberOfUrls));
+    const filterUrlsAsArray = Array.isArray(options.filterUrl)
+      ? options.filterUrl
+      : options.filterUrl
+      ? [options.filterUrl]
+      : [];
+    urls.push(...server.getAvailableUrls().filter(url => 
+      !filterUrlsAsArray.some(pattern => url.endsWith(":" + server.port + pattern))).slice(0, maxNumberOfUrls));
```

Comment:
let's split this out into some components rather than all one big line

```js
const availableUrls = server.getAvailableUrls();
const normalizedBlocklist = filterUrlsAsArray.map(...)
const urlsToUse = availableUrls.filter(url => !normalizedBlocklist.includes(url)).slice(0, maxNumberOfUrls)
```

> url.endsWith(":" + server.port + pattern))

we should reuse our logic below on 195-197 for consistency, not just appending port/pattern :)